VERSIONING.md

```diff
@@ -0,0 +1,221 @@
+Please see the [what to expect](#what-to-expect) section for more info.**
+
+<!-- MarkdownTOC autolink="true" style="ordered" indent="   " -->
+
+1. [Convention](#convention)
+1. [Public API](#public-api)
+   1. [Areas that *are* covered](#areas-that-are-covered)
+      1. [Intended for *public* consumption](#intended-for-public-consumption)
+      1. [Intended for *private* consumption](#intended-for-private-consumption)
+   1. [Areas that are *NOT* covered](#areas-that-are-not-covered)
+1. [Releases](#releases)
+   1. [Release channels](#release-channels)
+      1. [Stable channel](#stable-channel)
+      1. [Nightly channel](#nightly-channel)
+   1. [Release tracking](#release-tracking)
+      1. [Stable channel](#stable-channel-1)
+      1. [Nightly channel](#nightly-channel-1)
+   1. [Release downloading](#release-downloading)
+   1. [Release cadence](#release-cadence)
+      1. [Stable channel](#stable-channel-2)
+      1. [Nightly channel](#nightly-channel-2)
+1. [FAQ](#faq)
+   1. [Which release type should I be using?](#which-release-type-should-i-be-using)
+   1. [How does Vector treat patch and minor versions?](#how-does-vector-treat-patch-and-minor-versions)
+   1. [How does Vector treat major versions \(breaking changes\)?](#how-does-vector-treat-major-versions-breaking-changes)
+   1. [How does Vector treat pre-1.0 versions?](#how-does-vector-treat-pre-10-versions)
+   1. [Release cadence](#release-cadence-1)
+
+<!-- /MarkdownTOC -->
+
+## Convention
+
+Vector adheres to the [Semantic Versioning 2.0] convention. In summary:
+
+* Versions follow the `MAJOR.MINOR.PATCH` format (i.e., `2.5.1`)
+* `PATCH` increments only when backward compatible bug fixes are introduced.
+* `MINOR` increments only when new, backward compatible functionality is introduced.
+* `MAJOR` increments if any backwards incompatible changes are introduced.
+* Pre `1.0` (major version `0`) is for initial development and `MINOR` version bumps can introduce breaking changes.
+
+## Public API
+
+Semantic Versioning hinges on Vector's defintion of "public API". By nature of
+Vector - a tool that collects, processes, and routes data from disparate systems
+- it has a very large public surface area. It's not immediately obvious which
+parts are covered under our versioning contract and how they're covered. This
+section aims to remove all ambiguity in this area.
+
+### Areas that *are* covered
+
+The following Vector areas are covered in Vector's definition of public API.
+
+#### Intended for *public* consumption
+
+The follow Vector areas are inteded for *public* consumption (consumption by
+anything other than Vector itself). Backward incompatble changes will trigger
+a major version increment.
+
+* [CLI]
+  * The root [`vector` command], its input/output.
+  * The [`vector validate` subcommand], the exit codes.
+* [Data model]
+  * As exposed in the source of the [`lua` transform]
+  * As exposed in the output of the [`vector` sink]
```

Comment:
Is this true? We haven't publicly documented this. This seems like `Intended for *private* consumption` to me.

Reply:
Done in https://github.com/timberio/vector/pull/8795/commits/a22133309c8c08686c0e841c3053336ec970e8ff